diff --git a/node_modules/openai/core/error.js b/node_modules/openai/core/error.js
index 548c0c2..8f863b3 100644
--- a/node_modules/openai/core/error.js
+++ b/node_modules/openai/core/error.js
@@ -40,7 +40,9 @@ class APIError extends OpenAIError {
         if (!status || !headers) {
             return new APIConnectionError({ message, cause: (0, errors_1.castToError)(errorResponse) });
         }
-        const error = errorResponse?.['error'];
+        const error = errorResponse && typeof errorResponse === 'object' && 'error' in errorResponse ?
+            errorResponse['error']
+            : errorResponse;
         if (status === 400) {
             return new BadRequestError(status, error, message, headers);
         }
diff --git a/node_modules/openai/core/error.mjs b/node_modules/openai/core/error.mjs
index bfa952a..47bbdf5 100644
--- a/node_modules/openai/core/error.mjs
+++ b/node_modules/openai/core/error.mjs
@@ -36,7 +36,9 @@ export class APIError extends OpenAIError {
         if (!status || !headers) {
             return new APIConnectionError({ message, cause: castToError(errorResponse) });
         }
-        const error = errorResponse?.['error'];
+        const error = errorResponse && typeof errorResponse === 'object' && 'error' in errorResponse ?
+            errorResponse['error']
+            : errorResponse;
         if (status === 400) {
             return new BadRequestError(status, error, message, headers);
         }
diff --git a/node_modules/openai/src/core/error.ts b/node_modules/openai/src/core/error.ts
index d207856..e14045f 100644
--- a/node_modules/openai/src/core/error.ts
+++ b/node_modules/openai/src/core/error.ts
@@ -67,7 +67,10 @@ export class APIError<
       return new APIConnectionError({ message, cause: castToError(errorResponse) });
     }
 
-    const error = (errorResponse as Record<string, any>)?.['error'];
+    const error =
+      errorResponse && typeof errorResponse === 'object' && 'error' in errorResponse
+        ? (errorResponse as Record<string, any>)['error']
+        : errorResponse;
 
     if (status === 400) {
       return new BadRequestError(status, error, message, headers);
